### ISSUE TITLE
11.1.0-beta1 stackoverflow on property set (SetProperty), works in 11.0.10

### ISSUE THREAD

**Issue by BobbyCannon** [AvaloniaUI/Avalonia#15201]
### Describe the bug

This code below is stack overflowing on 11.1.0-beta1. If I roll back to 11.0.10 the code works fine.

``` CSharp
[Required(ErrorMessage = "Name is required!")]
public string Name
{
	get => _name;
	set => SetProperty(ref _name, value, true);
}
```

### To Reproduce

Call ObservableValidator.SetProperty in 11.1.0-beta1 leads to stackoverflow.

``` CSharp
[Required(ErrorMessage = "Name is required!")]
```

I think it's due to the required attribute but I'm not sure.

### Expected behavior

The code should not stackoverflow?

### Avalonia version

11.1.0-beta1

### OS

Windows

**Comment by maxkatz6** [comment]
Stacktrace (of stackoverflow) and SetProperty implementation are required to help here. Or minimal repro project.

**Comment by laolarou726** [comment]
I have the same issue, check if your App.axaml.cs's `OnFrameworkInitializationCompleted()` has following code:

```csharp
BindingPlugins.DataValidators.RemoveAt(0);
```

**Comment by maxkatz6** [comment]
@laolarou726 what exactly should be added to the release message?

**Comment by laolarou726** [comment]
> @laolarou726 what exactly should be added to the release message?

I mean this line of code. When I upgrading the Avalonia version for my older project. I also noticed this issue. After some investigation, I noticed this line of code has been added to the latest template project. So maybe add this notice some where in the document or Release message might be a good idea?

**Comment by timunie** [comment]
minimal repro to paste into `src/Samples/Sandbox`

```cs
        public MainWindow()
        {
            InitializeComponent();
            DataContext = new ViewModel();
        }

....

    public partial class ViewModel : ObservableValidator
    {
        // [Required(ErrorMessage = "Name is required!")] 
        [ObservableProperty] 
        [NotifyDataErrorInfo]
        private string? _Name;
    }
```

and 

```xml
  <TextBox Text="{Binding Name}" />
```

--------

seems to stuck around here: 
![image](https://github.com/AvaloniaUI/Avalonia/assets/47110241/ec21199b-5e7c-4c87-a073-2deabfa332bc)

**Comment by timunie** [comment]
... and here: 
![image](https://github.com/AvaloniaUI/Avalonia/assets/47110241/1ae23425-95cb-4f38-8148-349ee81029db)

**Comment by BobbyCannon** [comment]
I had commented that line of code then removed it.

![image](https://github.com/AvaloniaUI/Avalonia/assets/840590/d07e04bb-3eb8-45b8-b65b-10d2050edd23)

I was concerned about two things...
1. Why 0 index, what it that changes in an upgrade?
2. What was being removed.

This morning I wrote this...

``` CSharp
// Removes {Avalonia.Data.Core.Plugins.DataAnnotationsValidationPlugin}
var found = BindingPlugins.DataValidators.FirstOrDefault(x => x is DataAnnotationsValidationPlugin);
if (found != null)
{
	BindingPlugins.DataValidators.Remove(found);
}
```

It won't break if the validator moves to a different location.

Also is this still just a bug with the DataAnnotationsValidationPlugin?

I noticed while debugging this value adds { Value: } around the previous value.
